src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -967,6 +972,99 @@ public int compareTo(CachedAddresses other) {
             // ties are broken using unique id
             return Long.compare(this.id, other.id);
         }
+
+        /**
+         * Checks if the current cache record is expired or not. Expired records
+         * are removed from the expirySet and cache.
+         */
+        public boolean expired(long now) {
```

Comment:
This method's name does not reflect what this method does. It may be worth splitting.

Reply:
It is renamed to describe what it tries to do.